certification/internal/shell/runs_as_nonroot.go

```diff
@@ -1,37 +1,52 @@
 package shell
 
 import (
-	"bytes"
-	"os/exec"
 	"strconv"
 	"strings"
 
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
 	log "github.com/sirupsen/logrus"
 )
 
 type RunAsNonRootCheck struct{}
 
 func (p *RunAsNonRootCheck) Validate(image string) (bool, error) {
-	cmd := exec.Command("podman", "run", "-it", "--rm", "--entrypoint", "id", image, "-u")
-	var out bytes.Buffer
-	cmd.Stdout = &out
-	var stderr bytes.Buffer
-	cmd.Stderr = &stderr
-	err := cmd.Run()
+
+	line, err := p.getDataToValidate(image)
 	if err != nil {
-		log.Error("unable to get the id of the runtime user of this image")
-		log.Debugf("stdout: %s", out.String())
-		log.Debugf("stderr: %s", stderr.String())
 		return false, err
 	}
 
+	return p.validate(line)
+
+}
+
+func (p *RunAsNonRootCheck) getDataToValidate(image string) (string, error) {
+	runOpts := cli.ImageRunOptions{
+		EntryPoint:     "id",
+		EntryPointArgs: []string{"-u"},
+		LogLevel:       "debug",
+		Image:          image,
+	}
+	runReport, err := podmanEngine.Run(runOpts)
+	if err != nil {
+		log.Error("unable to get the id of the runtime user of this image, error: ", err)
+		log.Debugf("stdout: %s", runReport.Stdout)
+		log.Debugf("stderr: %s", runReport.Stderr)
+		return "", err
```

Comment:
better to return `nil` rather than "".

Reply:
Since the return type is just string instead of []string, you can't return nil.